### PR TITLE
OPSEXP-1826 Have a dedicated value to toggle outbound email configuration

### LINKED ISSUE #800: Outbound SMTP subsystem not configured if email.server.enabled not set to true
In the current Helm chart state, it is impossible to configure the outbound SMTP capability unless one also enables the inbound SMTP server, as in the config-repository file the outbound config properties are included in the conditional block guarded by `email.server.enabled` (beginning in line 91, see https://github.com/Alfresco/acs-deployment/blob/master/helm/alfresco-content-services/templates/config-repository.yaml#L91)

### PR DESCRIPTION
**Pull request by gionn**
Ref: OPSEXP-1826
Fix #800

### CODE REVIEW

File: helm/alfresco-content-services/templates/deployment-repository.yaml [hunk 1]
```diff
@@ -56,9 +56,8 @@ spec:
           - secretRef:
               name: {{ default (printf "%s-s3secret" (include "content-services.shortname" .)) .Values.s3connector.existingSecretName }}
           {{- end }}
-          {{- if .Values.email.server.enabled }}
+          {{- if .Values.mail.enabled }}
           - secretRef:
-              name: {{ template "content-services.shortname" . }}-mail-password
```
Comment (gionn) [inline helm/alfresco-content-services/templates/deployment-repository.yaml:61]: 😅

File: helm/alfresco-content-services/templates/deployment-repository.yaml [hunk 2]
```diff
@@ -56,9 +56,8 @@ spec:
...
               name: {{ default (printf "%s-mail-password" (include "content-services.shortname" .)) .Values.mail.existingSecretName }}
```
Comment (gionn) [inline helm/alfresco-content-services/templates/deployment-repository.yaml:61]: 😅

**Review by alxgomz — CHANGES_REQUESTED** [review]

File: helm/alfresco-content-services/templates/config-repository.yaml [hunk 3]
```diff
@@ -107,6 +94,21 @@ data:
       -Djavax.net.ssl.keyStore=/var/run/secrets/java.io/keystores/keystore.jks -Djavax.net.ssl.keyStorePassword=changeit
       {{- end }}
       {{- end }}
+      {{- if .Values.mail.host }}
+      -Dmail.encoding={{ .Values.mail.encoding }}
+      -Dmail.host={{ .Values.mail.host }}
+      -Dmail.port={{ .Values.mail.port }}
+      -Dmail.protocol={{ .Values.mail.protocol }}
+      -Dmail.username={{ .Values.mail.username }}
```
Comment (alxgomz) [inline helm/alfresco-content-services/templates/config-repository.yaml:102]:
```suggestion
      {{- printf('-Dmail.username=%s' .Values.mail.username) | default "" }}
```

As we can't really put a default inthe `values.yaml` file

File: helm/alfresco-content-services/templates/config-repository.yaml [hunk 4]
```diff
@@ -107,6 +94,21 @@ data:
...
+      -Dmail.from.default={{ .Values.mail.from.default }}
```
Comment (alxgomz) [inline helm/alfresco-content-services/templates/config-repository.yaml:103]:
```suggestion
      {{- printf('-Dmail.from.default=%s' .Values.mail.from.default) | default "" }}
```

**Review by alxgomz — APPROVED** [review]
please review suggestion otherwsie that's GTM

**Comment by gionn** [comment]
> please review suggestion otherwsie that's GTM

I understand that always explicitly providing a system prop via `-D` could actually change default behavior by overriding a possible default at application level, but I would prefer to not increase complexity if we don't have really an issue on a particular property.

**Pull request merged**